code/obj/item/assembly/misc_assemblies.dm

```diff
@@ -882,29 +882,40 @@ obj/item/assembly/radio_horn/receive_signal()
 	var/thingsneeded = null
 	var/obj/item/ammo/bullets/result = null
 	var/obj/item/accepteditem = null
+	var/craftname = null
 
 	proc/craftwith(obj/item/craftingitem, obj/item/frame, mob/user)
+
+		var/obj/item/material = null
 		if (istype(craftingitem, accepteditem)) //success! items match
-			src.thingsneeded --
+			var/consumed = min(src.thingsneeded, craftingitem.amount)
+
+			//material = craftingitem
+			thingsneeded -= consumed
+			boutput(user, "consuming [consumed] things!")
```

Comment:
Was this a debug statement left behind? This seems very weird to show to a person.